src/webapp/default_settings.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+
+"""
+Flask environment configuration
+"""
+
+from environs import Env
+
+env = Env()
+env.read_env()
+
+ENV = env.str("FLASK_ENV", default="production")
+DEBUG = True
+
+# For local development purposes
+SESSION_TYPE = "filesystem"
+
+# Azure AD configuration - Update these three to your values
+
+# This application's "Application (client) ID" found in this application's
+# app registration in Azure Active Directory
+# Example: "8a193f2d-e89d-c577-81b7-145f76344a77"
+CLIENT_ID = ""
+
+# Any of the non-expired "Client secrets" values found in this application's
+# app registration in Azure Active Directory
+# Example: "jPk7Q~NeKvlrrCj2_0tl~q-PR~o9Tp10W88"
+CLIENT_CREDENTIAL = ""
+
+# The tenant's "Primary domain" in which this application was registered,
+# prefixed with https://login.microsoftonline.com/
+# Example: "https://login.microsoftonline.com/contoso.onmicrosoft.com"
+AUTHORITY = ""
```

Comment:
in other samples, we are already asking for `Tenant Id` instead. I think it'd be great to arrive to a convention provided both are valid options.

Reply:
Yeah, good point.  I honestly wish the python msal libraries took just tenant ID in this case.  I'll leave this comment open for others to provide an opinion on.

Reply:
Most of the samples I've seen construct the authority URI from "cloud instance" and tenant ID. Since the cloud instance can change based on several factors, I'd stick with that.

Reply:
I'm sorry, stick with `https://login.microsoftonline.com/contoso.onmicrosoft.com` or stick with `contoso` and hardcode the `login.microsoftonline.com/` prefix in the code?

Reply:
The "that" in "stick with that" being what the other samples do which is to construct the full authority URI from two user-specified vars, _cloud instance_ and _tenant ID_.

Am I reading right, though, that MSAL Python differs on this?

Reply:
Hmmm... authority is a tough one:

https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-client-application-configuration#authority

Reply:
I don't know what a "cloud instance" is -- authority is as you provided the link, simply the constant prefix of `https://login.microsoftonline.com/` and then either `common`, `organizations`, `consumers`, or best, your tenant ID for single tenant applications.  That full URL is the authority.  So when I have a property called `AUTHORITY`, i'd expect it, as a user to match what an authority is defined as.  If instead you wanted me to call that `AUTHORITY_SUFFIX` or `TENANT`, I could change it and then concat that prefix with that suffix in the code.

This isn't unique to python msal, hence my confusion at this.  Just want to know which way you'd like me to go. Thanks

Reply:
Well, apparently, there is no standardized format. Roll with whatever format makes sense for the sample.

Variably, we currently do this:

- Hardcode the full authority value
- Hardcode part of the authority value
- Construct the authority value from variables

Some examples:

[ASP.NET Core quickstart](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-v2-aspnet-core-webapp#register-and-download-the-app):

```json
"Domain": "[Enter the domain of your tenant, e.g. contoso.onmicrosoft.com]",
"ClientId": "Enter_the_Application_Id_here",
"TenantId": "common",
```
[Node.js console app tutorial](https://docs.microsoft.com/en-us/azure/active-directory/develop/tutorial-v2-nodejs-console):

```javascript
const msalConfig = {
    auth: {
        clientId: process.env.CLIENT_ID,
        authority: process.env.AAD_ENDPOINT + '/' + process.env.TENANT_ID,
        clientSecret: process.env.CLIENT_SECRET,
    }
};
```

[Python console quickstart](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-v2-python-daemon):

```json
"authority": "https://login.microsoftonline.com/Enter_the_Tenant_Id_Here",
"client_id": "Enter_the_Application_Id_Here",
"secret": "Enter_the_Client_Secret_Here"
```

[iOS/macOS tutorial](https://docs.microsoft.com/en-us/azure/active-directory/develop/tutorial-v2-ios):

```swift
// Update the below to your client ID you received in the portal. The below is for running the demo only
let kClientID = "Your_Application_Id_Here"
let kGraphEndpoint = "https://graph.microsoft.com/" // the Microsoft Graph endpoint
let kAuthority = "https://login.microsoftonline.com/common" // this authority allows a personal Microsoft account and a work or school account in any organization's Azure AD tenant to sign in
```

Reply:
Sounds good, thanks for looking it up.  If a "natural" method emerges across all of the samples we do here, from a content development/story-telling perspective, we can always go back and unify them.  Going to mark resolved for now.